### PR TITLE
Fix direct chat issues #706

### PR DESCRIPTION
**Pull request by pedroGitt**

### CODE REVIEW

File: vector/src/main/java/im/vector/activity/CommonActivityUtils.java [hunk 1]
```diff
@@ -1156,6 +1157,32 @@ public static boolean isOneToOneRoomJoinedByUserId(final Room aRoom, final Strin
...
+            if(!aParticipantUserId.equals(aSession.getMyUserId())) {
+                params.put("invite", Arrays.asList(aParticipantUserId));
+            }
+
+            // Direct message case
+            aSession.createRoom(params, aCreateRoomCallBack);
+        }
+
+        return retCode;
+    }
```
Comment (ylecollen) [inline vector/src/main/java/im/vector/activity/CommonActivityUtils.java]: This function is not anymore used (like createDirectMessagesRoom)

File: vector/src/main/java/im/vector/activity/CommonActivityUtils.java [hunk 2]
```diff
@@ -1267,45 +1294,48 @@ public void onUnexpectedError(Exception e) {
...
-                    CommonActivityUtils.goToRoomPage(fromActivity, aSession, params);
+                    if(isGoToRoomPage) {
+                        HashMap<String, Object> params = new HashMap<>();
+                        params.put(VectorRoomActivity.EXTRA_MATRIX_ID, aSession.getMyUserId());
+                        params.put(VectorRoomActivity.EXTRA_ROOM_ID, aRoomId);
+                        params.put(VectorRoomActivity.EXTRA_EXPAND_ROOM_HEADER, true);
+
+                        Log.d(LOG_TAG, "## setToggleDirectMessageRoom(): invite() onSuccess - start goToRoomPage");
+                        CommonActivityUtils.goToRoomPage(fromActivity, aSession, params);
+                    }
```
Comment (ylecollen) [inline vector/src/main/java/im/vector/activity/CommonActivityUtils.java]: the true value is not anymore set.

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java [hunk 3]
```diff
@@ -2518,11 +2543,37 @@ private void onJoined() {
...
+
+            if(2==members.size()) {
+                String participantUserId;
+
+                for (RoomMember member : members) {
+                    // search for the second participant
+                    if (!member.getUserId().equals(myUserId)) {
+                        participantUserId = member.getUserId();
+                        CommonActivityUtils.setToggleDirectMessageRoom(mSession, sRoomPreviewData.getRoomId(), participantUserId, this, mDirectMessageListener, false);
+                        break;
```
Comment (ylecollen) [inline vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: check if the room is already a direct message one.

**Comment by pedroGitt** [comment]
remarks done

**Comment by ylecollen** [comment]
LGTM

**Pull request merged**